### PR TITLE
Medium sized tests

### PR DESCRIPTION
**Pull request by Legersky**
The goal of this PR is to run tests that take a bit longer only when merging to main.

### CODE REVIEW

File: pyproject.toml
Comment (matthiashimmelmann) [inline pyproject.toml:1]: Doesn't `medium` ideologically replace `slow`? Should we turn `slow` into `long`? Or should we do that on a case-by-case basis as is done in `realization_counting`?

Comment (matthiashimmelmann) [reply pyproject.toml:1]: Or we do it the other way round, call `medium` `slow` instead and replace long with `unsuitable_for_automatic_testing` or `infeasibly_large` to highlight that these tests are not automatically run?

**Review by matthiashimmelmann — COMMENTED** [review]
I agree that `medium` is not ideal. Maybe we can be more descriptive here.`tests_for_dev`, `slow_tests_for_main` and `local_tests` would be another option.

**Comment by Legersky** [comment]
`realization_counting` is different in the sense that it tests functionality requiring an optional package.

I like the idea of indicating when they are run, what about the following:
```
markers = [
    "slow_main: marks slower tests run at merge to main",
    "long_local: marks long tests run locally",
    "realization_counting: marks tests using extra realization-counting",
    "meshing: marks tests using extra meshing"
]
```
?

**Comment by matthiashimmelmann** [comment]
> `realization_counting` is different in the sense that it tests functionality requiring an optional package.
> 
> I like the idea of indicating when they are run, what about the following:
> 
> ```
> markers = [
>     "slow_main: marks slower tests run at merge to main",
>     "long_local: marks long tests run locally",
>     "realization_counting: marks tests using extra realization-counting",
>     "meshing: marks tests using extra meshing"
> ]
> ```
> 
> ?

I like it!

**Review by matthiashimmelmann — APPROVED** [review]

**Pull request merged**